### PR TITLE
feat: add inverse-solution API and shared angle solver

### PR DESCRIPTION
**Pull request by joelbenway**
Summary
Adds an inverse-solution API to lob — given a target range and measured impact, compute the required launch-angle and windage adjustments in MOA — backed by a shared angle solver.
Highlights
- New C/C++ API (include/lob/lob.h, lob.hpp):
- LobFastInverse / LobFastInverse — fast approximate inverse using the vacuum model
- LobSolveInverse / LobSolveInverse — full solve via the shared root-finder
- Shared angle solver (source/solve_angle.hpp): single consolidated secant-based root-finder for inverse solutions and zero-angle determination, removing ~50 lines of duplicated logic from source/lob_builder.cpp.
- Error handling: dedicated kLobErrorZeroUnreachable when a target lies beyond ballistic reach.
- Physics fixes: corrected vacuum seed formula and standardized gravity constant to 32.17405 ft/s².
Accuracy fixes (from pre-merge review)
- fix(solve): inverse solver now seeds with the mechanical zero angle; SolveAngle adds aerodynamic_jump internally, so the previous seed double-counted it on the first iteration.
- perf(solve): hoisted drag_curve out of the iteration loop and bail out early when residuals go non-finite.
Testing
- New suites: lob_inverse_test.cpp, solve_angle_test.cpp — 230 + 283 lines covering physical invariants, bounds, invalid contexts, and pointer/array overloads.
- Full suite: 501/501 passing.
Benchmark
Cachegrind benchmark updated for the new API.

## Summary by CodeRabbit

- **New Features**
  - Added inverse ballistic solving to calculate launch adjustments in MOA.
  - Added fast conversion of forward-solution results to inverse adjustments.
  - Added overloads for single values, arrays, and pointer-based inputs.
  - Added an error status for unreachable zero-angle solutions.

- **Bug Fixes**
  - Improved input validation, trajectory handling, angle solving, and gravity precision.

- **Documentation**
  - Clarified forward and inverse result meanings and builder usage.

- **Tests**
  - Added comprehensive coverage for inverse solving and edge cases.

### CODE REVIEW

File: include/lob/lob.h [hunk 1]
```diff
@@ -89,6 +89,7 @@ enum {
   kLobErrorZeroAngleOOR,
   kLobErrorZeroDataRequired,
   kLobErrorZeroDistanceOOR,
+  kLobErrorZeroUnreachable,
```
[the remark this answers is not in the document]
Comment (joelbenway) [reply include/lob/lob.h:92]: Breaking compatibility doesn't matter at this point as there are no users. Keep alphabetical order.

File: source/lob_solve.cpp [hunk 2]
```diff
@@ -153,6 +149,62 @@ size_t LobSolve(const LobContext* pctx, const uint32_t* pranges,
...
+    }
+    const FeetT kElevation = InchT(pouts[i].elevation);
+    const FeetT kRange = FeetT(pouts[i].range);
+    const RadiansT kSeed = FastInverseAngle(kSeedBase, kElevation, kRange);
+    const RadiansT kTheta = SolveAngle(*pctx, kRange, FeetT(0.0), kSeed);
+    if (kTheta.IsNaN()) {
+      return i;
+    }
+    pouts[i].elevation = MoaT(kTheta - kZeroAngle).Value();
+    pouts[i].deflection = LobInchToMoa(-pouts[i].deflection, kRange.Value());
```
[the remark this answers is not in the document]
Comment (joelbenway) [reply source/lob_solve.cpp:203]: This isn't helpful. The forward solver already aborts early when a range is unreachable so kForwardSolves will be < size. Only reachable trajectories will be worked on in the for loop.

File: SECURITY.md [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+## What to do
+
+Report security issues with [GitHub's private vulnerability reporting](https://github.com/joelbenway/lob/security/advisories/new) rather than a public issue. If private reporting isn't available for the repository, [raise an issue](https://github.com/joelbenway/lob/issues) and note it as a security report.
+
+Please do not disclose the vulnerability publicly until I've had a chance to triage it. Whatever channel you use, be generous with detail: version and commit, platform and compiler, and a minimal reproduction. A good report is worth a lot more than a good fix.
+
+## What happens next
+
+Lob is a solo-authored project, but rest assured, our entire security department will work tirelessly to acknowledge a report within a few days. I'll post updates as the investigation proceeds. If I'm too quiet for your tastes, feel free to poke me.
```
[the remark this answers is not in the document]
Comment (joelbenway) [reply SECURITY.md:13]: no, it's a joke
